testing/e2e/app/routes/ToggleSwitch/route.tsx

```diff
@@ -1,27 +1,40 @@
-import { ToggleSwitch } from '@itwin/itwinui-react';
+import { ThemeProvider, ToggleSwitch } from '@itwin/itwinui-react';
 import { useSearchParams } from 'react-router';
 
 export default function Page() {
   const [searchParams] = useSearchParams();
+
+  const themeProviderConsistentPropsSpread =
+    searchParams.get('themeProviderConsistentPropsSpread') != null
+      ? searchParams.get('themeProviderConsistentPropsSpread') === 'true'
+      : undefined;
   const shouldPassWrapperProps =
     searchParams.get('shouldPassWrapperProps') === 'true';
 
   return (
-    <ToggleSwitch
-      className='my-class'
-      style={{
-        backgroundColor: 'red',
+    <ThemeProvider
+      future={{
+        ToggleSwitch: {
+          consistentPropsSpread: themeProviderConsistentPropsSpread,
+        },
```

Comment:
As this is now adding an extra ThemeProvider, I think this entire test case should be moved into a new component that gets conditionally rendered depending on searchparams. This way, the other test cases are unaffected.

Reply:
I think you misunderstood. I'm suggesting creating a completely separate branch (early return), not just a wrapper around the existing branch. See `Popover/route.tsx` or `DropdownMenu/route.tsx` for an example. This makes the tests independent of each other.